### PR TITLE
Issue-33: made the replay extension work

### PR DESCRIPTION
**Pull request by kaczenski**
We investigated why the replay extension did not work out of the box and came across the examples in java and javascript:
https://developer.salesforce.com/docs/atlas.en-us.api_streaming.meta/api_streaming/code_sample_generic_vfp_walkthrough.htm
and 
https://github.com/developerforce/StreamingReplayClientExtensions/blob/master/javascript/cometdReplayExtension.js

and we noticed the extension is set to other channels (e.g. /event/subscribe/) and the message body also contains the channel id we would like to subscribe to ("ext : { "replay" : { CHANNEL : REPLAY_VALUE }}".

Resources: https://developer.salesforce.com/docs/atlas.en-us.api_streaming.meta/api_streaming/using_streaming_api_durability.htm?search_text=replay

We are aware this could be a BC change but think their shouldn't be too many folks having implemented this extension because it does not seem to work. If so, I'd like to hear about their opinion about this and how they made it work...

### CODE REVIEW

**Comment by curiosity26** [comment]
According to the cometD example you've provided, the replayId is only applied to an outgoing message when subscribing to the channel. At that time, the message's channel is `/meta/subscribe`, all other messages after subscription will have be based upon the replayId given at subscription and have their own replayId in the payload of the message.

https://github.com/developerforce/StreamingReplayClientExtensions/blob/master/javascript/cometdReplayExtension.js#L59

In short, we only send the replayID in the outgoing message to Salesforce on `/meta/subscription` requests.

**Comment by curiosity26** [comment]
Also I feel I should clarify a slight difference that may change in the future. The Java client applies extensions, like the Replay Extension, to the TopicSubscriber, which has a direct relationship with the channel. 

In the case of this SDK, we don't have TopicSubscribers, mainly because they make great sense in a Threadable language, but they don't work too well when translated to PHP. For this reason, all extensions are applied to BayuexClient and then applied to each subscription request when the client starts.

In the future, I'll see about maybe allowing channels to have their own set of extensions that can be applied at different times in the streaming lifecycle or perhaps filtering out which channels a given extension applies to. At this time, that change will not be made and we will continue to use the shared extensions pattern, but I will add it into the issues queue as an enhancement.

Thank you for taking the time to share your code with us and bring this to our attention. Let us know if anything comes up in the future.

**Comment by curiosity26** [comment]
@kaczenski Just wanted to give you an update. We did confirm there was an issue with the ReplayExtension and a fix has now been pushed.

The ReplayExtension still is applied to all channels but in the future we will allow specific channels to use independent replayId's as this is necessary when specifying a particular replayId for a given channel.

Thanks
